go/bazel.go

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2017 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bazel_go
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+var _ = fmt.Printf
+
+var bazelVersion = "bazel_version was not set. This is an error."
+
+type TestingBazel interface {
+	ScratchDir(name string) error
```

Comment:
Why provide these methods in a public interface? Seems like they would always have the same implementation, and they don't depend on any state. Can they be functions instead? Possibly private, if they're only used in bazel_test.go?

Reply:
These have been moved to be static methods. I attempted to make bazel_test.go effectively an external test. I have made that more formal in this commit.